### ISSUE TITLE
interface-over-type-literal triggered even though type cannot be expressed as an interface

### ISSUE THREAD

**Issue by bschlenk** [palantir/tslint#3717]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.6.2
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
export type MyType = { [key: number]: boolean };
```

with `tslint.json` configuration:

```json
{
    "extends": [
        "tslint:recommended",
        "tslint-config-airbnb"
    ],
    "rules": {
        "import-name": false,
        "no-increment-decrement": false,
        "no-this-assignment": false,
        "no-conditional-assignment": false,
        "interface-name": false
    }
}
```

#### Actual behavior
TSLint complains that this code snippet violates interface-over-type-literal. But I don't see a way to express a map of arbitrary numbers to booleans with an interface.

#### Expected behavior
This should be perfectly acceptable code since there is no way to express this as an interface. Or, Perhaps the warning should tell me that using a Map object is a better solution?
